### ISSUE TITLE
Info log messages ignore level filtering

### ISSUE THREAD

**Issue by Kolyunya** [yiisoft/yii2#16251]
### What steps will reproduce the problem?
Implement a custom log target. Configure it to receive only `error` messages. Produce a single `error` message. Inspect `$this->messages` from the `export()` method of the custom log target.

### What is the expected result?
A single `error` message.

### What do you get instead?
Two messages. An expected `error` message and another `info` message with the request context.

### Additional info
This is caused by [this code block](https://github.com/yiisoft/yii2/blob/master/framework/log/Target.php#L127-L129). Not sure why it's there but it's a very annoying and counter-intuitive behavior. I explicitly configured the target to receive only `error` messages and I do not want to filter them _again_ manually.

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15.1
| PHP version      | 7.1
| Operating system | GNU/Linux

**Comment by rob006** [comment]
I don't think that this could be fixed in 2.0. This is more like a design issue - fake context message should not be injected at `collect()` level. This approach may be fine for `FileTarget`, but it does not fit well to other targets. Context message should handled at `export()` level and adapted to specific target. 

For now, if you don't want to log context message, you can set `$logVars` to empty array.
